src/main/java/com/owncloud/android/ui/asynctasks/LoadUrlTask.java

```diff
@@ -22,42 +22,42 @@
 
 import android.accounts.Account;
 import android.os.AsyncTask;
-import android.os.Build;
-import android.os.Handler;
-import android.view.View;
-import android.widget.Toast;
 
-import com.google.android.material.snackbar.Snackbar;
-import com.owncloud.android.R;
+import com.nextcloud.android.lib.resources.directediting.DirectEditingOpenFileRemoteOperation;
 import com.owncloud.android.lib.common.operations.RemoteOperationResult;
 import com.owncloud.android.operations.RichDocumentsUrlOperation;
-import com.owncloud.android.ui.activity.RichDocumentsWebView;
-import com.owncloud.android.utils.DisplayUtils;
-import com.owncloud.android.utils.ThemeUtils;
+import com.owncloud.android.ui.activity.EditorWebView;
+import com.owncloud.android.ui.activity.RichDocumentsEditorWebView;
+import com.owncloud.android.ui.activity.TextEditorWebView;
 
 import java.lang.ref.WeakReference;
 
-import androidx.annotation.RequiresApi;
-
-@RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
 public class LoadUrlTask extends AsyncTask<String, Void, String> {
 
     private Account account;
-    private WeakReference<RichDocumentsWebView> richDocumentsWebViewWeakReference;
+    private WeakReference<EditorWebView> editorWebViewWeakReference;
 
-    public LoadUrlTask(RichDocumentsWebView richDocumentsWebView, Account account) {
+    public LoadUrlTask(EditorWebView editorWebView, Account account) {
         this.account = account;
-        this.richDocumentsWebViewWeakReference = new WeakReference<>(richDocumentsWebView);
+        this.editorWebViewWeakReference = new WeakReference<>(editorWebView);
     }
 
     @Override
     protected String doInBackground(String... fileId) {
-        if (richDocumentsWebViewWeakReference.get() == null) {
+        if (editorWebViewWeakReference.get() == null) {
+            return "";
+        }
+
+        RemoteOperationResult result = null;
+        if (editorWebViewWeakReference.get() instanceof RichDocumentsEditorWebView) {
```

Comment:
editorWebViewWeakReference can become `null` between line 52 and 53.

```suggestion
        final editorWebViewWeak = editorWebViewWeakReference.get(); // lock reference so it's not GC'd between lines
        if (editorWebViewWeak == null) { 
            return;
        }
        and so on
```
Just discovered suggestions feature. Neat!